src/Microsoft.DotNet.Wpf/src/Themes/PresentationFramework.Fluent/Styles/DataGrid.xaml

```diff
@@ -15,394 +15,568 @@
+                    </ControlTemplate.Triggers>
                 </ControlTemplate>
             </Setter.Value>
         </Setter>
+        <Style.Triggers>
+            <Trigger Property="IsNewItem" Value="True">
+                <Setter Property="Margin"
+                    Value="{Binding RelativeSource={RelativeSource AncestorType={x:Type DataGrid}}, Path=NewItemMargin}" />
+            </Trigger>
+        </Style.Triggers>
     </Style>
 
-    <!--  Style and template for the resize control on the DataGridRowHeader.  -->
-    <Style x:Key="RowHeaderGripperStyle" TargetType="{x:Type Thumb}">
-        <Setter Property="Height" Value="8" />
-        <Setter Property="Background" Value="Transparent" />
-        <Setter Property="Cursor" Value="SizeNS" />
+    <Style x:Key="DefaultDataGridCellStyle" TargetType="{x:Type DataGridCell}">
+        <Setter Property="Background" Value="{DynamicResource DataGridCellBackground}" />
+        <Setter Property="BorderBrush" Value="{DynamicResource DataGridCellBorderBrush}" />
+        <Setter Property="BorderThickness" Value="{DynamicResource DataGridCellBorderThickness}" />
+        <Setter Property="FocusVisualStyle" Value="{StaticResource DataGridCellFocusVisualStyle}" />
+        <Setter Property="MinHeight" Value="{StaticResource DataGridCellMinHeight}" />
         <Setter Property="OverridesDefaultStyle" Value="True" />
         <Setter Property="Template">
             <Setter.Value>
-                <ControlTemplate TargetType="{x:Type Thumb}">
-                    <Border Padding="{TemplateBinding Padding}" Background="{TemplateBinding Background}" />
+                <ControlTemplate TargetType="{x:Type DataGridCell}">
+                    <Border x:Name="RootBorder"
+                        Background="{TemplateBinding Background}"
+                        BorderBrush="{TemplateBinding BorderBrush}"
+                        BorderThickness="{TemplateBinding BorderThickness}"
+                        SnapsToDevicePixels="True">
+                        <ContentPresenter
+                            SnapsToDevicePixels="{TemplateBinding SnapsToDevicePixels}"
+                            VerticalAlignment="Stretch"
+                            Margin="{TemplateBinding Padding}" />
+                    </Border>
+                    <ControlTemplate.Triggers>
+                        <Trigger Property="IsMouseOver" Value="True">
+                            <Setter Property="Background"
+                                Value="{DynamicResource DataGridCellBackgroundPointerOver}" />
+                            <Setter Property="BorderBrush"
+                                Value="{DynamicResource DataGridCellBorderBrushPointerOver}" />
+                            <Setter Property="Foreground"
+                                Value="{DynamicResource DataGridCellForegroundPointerOver}" />
+                        </Trigger>
+                        <Trigger Property="IsSelected" Value="True">
```

Comment:
Should we have cornerradius (e.g. 2px?) when focused? E.g. Loop:
 
![image](https://github.com/user-attachments/assets/f28ce36c-fdf4-409c-9a39-1bcb5ba8a209)

Reply:
Nit: Maybe also a small (left) padding for each cell / header item? Should be overrideable though